### ISSUE TITLE
Feature: Support to Enumerable Scalar Values

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#538]
Currently, RepoDB requires a concreate .NET CLR Type return type just to retrieve the results from the fetch operations (i.e: [ExecuteQuery](https://repodb.net/operation/executequery), [Query](https://repodb.net/operation/query), [QueryAll](https://repodb.net/operation/queryall) and [BatchQuery](https://repodb.net/operation/batchquery)).

Let us say, you only would like to query all the product name from the `[dbo].[Product]` table (like below).

```csharp
using (var connection = new SqlConnection(connectionString))
{
	var result = connection.ExecuteQuery<ProductNameOnly>("SELECT Name [dbo].[Product] ORDER BY Name;");
}
```

You are required to create a concrete class `ProductNameOnly`.

```csharp
public class ProductNameOnly
{
	public string Name { get; set; }
}
```

This is quite too extensive in the client side (or the library consumer) just to create this un-organized class.

This user story will allow the developers to do the magic query of the targetted Scalar (column-based) values. The propose call would be below.

**String**:

```csharp
using (var connection = new SqlConnection(connectionString))
{
	var result = connection.ExecuteQuery<string>("SELECT Name [dbo].[Product] ORDER BY Name;");
}
```

**Long/Int**:

```csharp
using (var connection = new SqlConnection(connectionString))
{
	var result = connection.ExecuteQuery<long>("SELECT Id [dbo].[Order] ORDER BY Id;");
}
```

In which the result is of type `IEnumerable<Type>`.

**Comment by mikependon** [comment]
The fix is available at version [RepoDb v1.12.0-beta3](https://www.nuget.org/packages/RepoDb/1.12.0-beta3) or above.

**Comment by mikependon** [comment]
Closing this ticket now.
